plugins/parodos/src/components/workflow/workflowDetail/WorkFlowDetail.tsx

```diff
@@ -78,18 +78,21 @@ export const WorkFlowDetail = () => {
 
     const updateWorkflowExecutionState = async () => {
       const data = await fetch(`${workflowsUrl}/${executionId}/status`);
-
-      // TODO: so far failing, should be fixed by https://issues.redhat.com/browse/FLPATH-184
       const response = (await data.json()) as WorkflowStatus;
 
       return response.works;
     };
-    const taskInterval = setInterval(() => {
+
+    const updateWorksFromApi = () => {
       updateWorkflowExecutionState().then(fetchedTasks => {
         updateWorks(fetchedTasks);
       });
-    }, 5000);
+    };
 
+    const taskInterval = setInterval(() => {
```

Comment:
We will nee to introduce more generic way of polling.
In addition, it should include exclusivity of the call, for the case that the previous call takes long to fulfill.